packages/backend/src/peripherals/database/TxCountRepository.ts

```diff
@@ -0,0 +1,116 @@
+import { Logger } from '@l2beat/common'
+import { ProjectId, UnixTime } from '@l2beat/types'
+import { TxCountRow } from 'knex/types/tables'
+
+import { BaseRepository } from './shared/BaseRepository'
+import { Database } from './shared/Database'
+
+export interface TxCountRecord {
+  timestamp: UnixTime
+  projectId: ProjectId
+  blockNumber: number
+  count: number
+}
+
+interface MissingNumber {
+  i: number
+}
+
+export class TxCountRepository extends BaseRepository {
+  constructor(database: Database, logger: Logger) {
+    super(database, logger)
+
+    /* eslint-disable @typescript-eslint/unbound-method */
+
+    this.add = this.wrapAdd(this.add)
+    this.addMany = this.wrapAddMany(this.addMany)
+    this.deleteAll = this.wrapDelete(this.deleteAll)
+
+    /* eslint-enable @typescript-eslint/unbound-method */
+  }
+
+  async add(record: TxCountRecord) {
+    const knex = await this.knex()
+    const row = toRow(record)
+    await knex('tx_count').insert(row)
+    return `${row.project_id}-${row.block_number}`
+  }
+
+  async addMany(records: TxCountRecord[]) {
+    const knex = await this.knex()
+    const rows = records.map(toRow)
+    await knex('tx_count').insert(rows)
+    return rows.length
+  }
+
+  async findLatestByProject(projectId: ProjectId) {
+    const knex = await this.knex()
+    const row = await knex('tx_count')
+      .where('project_id', projectId.toString())
+      .orderBy('block_number', 'desc')
+      .first()
+
+    return row ? toRecord(row) : undefined
+  }
+
+  async getMissingByProject(projectId: ProjectId) {
```

Comment:
this method looks pretty complex, have you tested it with big dataset?

Reply:
Onnit